WooCommerce/Classes/POS/Presentation/Item Selector/ItemList.swift

```diff
@@ -3,18 +3,57 @@ import enum Yosemite.POSItem
 import struct Yosemite.POSVariableParentProduct
 
 /// Displays a list of POS items or placeholder card based on the given state.
-struct ItemList: View {
+struct ItemList<HeaderView: View>: View {
+    enum BaseItem {
+        case root
+        case parent(POSItem)
+    }
+
+    @Environment(\.floatingControlAreaSize) var floatingControlAreaSize: CGSize
+    @EnvironmentObject var posModel: PointOfSaleAggregateModel
     let state: ItemListState
+    private let node: BaseItem
+    private let headerView: HeaderView
+
+    init(state: ItemListState,
+         node: BaseItem = .root,
+         @ViewBuilder headerView: () -> HeaderView = { EmptyView() }) {
+        self.state = state
+        self.node = node
+        self.headerView = headerView()
+    }
 
     var body: some View {
-        ForEach(state.items) { item in
-            ItemListRow(item: item)
+        ScrollView {
+            LazyVStack {
+                headerView
+
+                ForEach(state.items) { item in
+                    ItemListRow(item: item)
+                }
+
+                switch state {
+                case .loading, .loaded(_, hasMoreItems: true):
+                    GhostItemCardView()
+                        .onAppear {
+                            guard case .root = node else { return }
+                            Task { await posModel.loadNextItems() }
```

Comment:
I'm guessing this is where we'll trigger the child item next page load as well. Do you plan to add a different `loadNextItems` given a parent node, or add a parameter to `loadNextItems` with the parent item?

Reply:
Yep – I'm intending to add a parameter for the parent item 😊